core/remote/watcher.js

```diff
@@ -318,19 +321,23 @@ class RemoteWatcher {
             continue
           } else {
             _.assign(previousChange, {
-              type: 'IgnoredChange',
+              type: 'DescendantChange',
               detail: `Folder was moved as descendant of ${change.doc.path}`
             })
+            // $FlowFixMe
+            remoteChange.includeDescendant(change, previousChange)
```

Comment:
Didn't we just lose `previousChange.descendantDirMoves`?

Reply:
My bad, we didn't, it's just that flow doesn't see them anymore (still an issue though).

So I suppose we could either:

1. progressively flatten `*.descendantDirMoves` (by concatenating them in `includeDescendant()`
2. flatten them all at once in `RemoteWatcher#apply()`

Reply:
Did :one: in #1274 / https://github.com/cozy-labs/cozy-desktop/pull/1274/commits/fc34fc7361795bac4348d7e6759748f82cefd78f